### PR TITLE
Update NvFuserOperatorSupport

### PR DESCRIPTION
**Pull request by SherlockNoMad**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #81311

### CODE REVIEW

**Review by davidberard98 — APPROVED** [review]
LGTM! But I did have a question about `square`

File: torch/fx/passes/backends/nvfuser.py [hunk 1]
```diff
@@ -187,31 +188,22 @@ def __init__(self):
             "torch.ops.aten.linear": None,
             "torch.ops.aten.gelu": None,
             "torch.ops.aten.gelu_backward": None,
+            # "torch.ops.aten.hardtanh": None,        # has functional ref, using unsupported aten.clamp
+            "torch.ops.aten.leaky_relu": None,
+            "torch.ops.aten.square": None,
```
[the remark this answers is not in the document]
Comment (SherlockNoMad) [reply torch/fx/passes/backends/nvfuser.py:193]: It does has ref impl... although it says aten_op=None. So I am not sure what this means... 

@mruberry 

```
@_make_elementwise_unary_reference(
    ELEMENTWISE_TYPE_PROMOTION_KIND.BOOL_TO_LONG,
    aten_op=None,  # CompositeImplicitAutograd,
)
def square(a: TensorLikeType) -> TensorLikeType:
    return mul(a, a)
```

[the remark this answers is not in the document]
Comment (mruberry) [reply torch/fx/passes/backends/nvfuser.py:193]: @ezyang should know what's up here. I expected the ATen FX graph we receive to have already traced through composite implicit autograd operators -- but maybe that's not the case?

**Pull request closed without merging**